.github/scripts/release.py

```diff
@@ -86,60 +110,148 @@ def printReleases(self):
+        # Overwrite existing tags
+        info('Pushing tag "%s"' % self.tag)
+        tag_info = self.local_repo.create_tag(self.tag, message=self.tag_msg, force=True)
+        self.local_repo.git.push(tags=True, force=True)
+
+    def deleteTag(self):
+        # Remove from remote
+        if self.tag in self.local_repo.tags:
+            info('Deleting tag "%s"' % self.tag)
+            self.local_repo.remote('origin').push(':%s' % self.tag)
+        else:
+            info('A tag does not exists for "%s". No need to delete.' % self.tag)
+
+    def updateSubmodulePointer(self, rel_path, ref):
+        submodule = Repo(rel_path)
+        submodule.remote('origin').fetch()
+        submodule.git.checkout(ref)
+
+    def updateFileHeaderVersions(self, old_version_substrings, new_version_string):
+        info('Updating file header versions for "%s"...' % self.version, end='')
+        n_updated = 0
+        n_updated += update_version_number_in_freertos_component(self.repo_path,
+                                                                 '.',
+                                                                 old_version_substrings,
+                                                                 new_version_string,
+                                                                 exclude_hidden=True)
+
+        n_updated += update_version_number_in_freertos_component(os.path.join('.github', 'scripts'),
+                                                                 self.repo_path,
+                                                                 old_version_substrings,
+                                                                 new_version_string,
+                                                                 exclude_hidden=False)
+
+        print('...%d Files updated.' % n_updated)
+
+        self.commitChanges(self.commit_msg_prefix + 'Bump file header version to "%s"' % self.version)
+
+    def deleteGitRelease(self):
+        info('Deleting git release endpoint for "%s"' % self.tag)
+
+        try:
+            self.repo.get_release(self.tag).delete_release()
+        except UnknownObjectException:
+            info('A release endpoint does not exist for "%s". No need to erase.' % self.tag)
+        except:
+            assert False, 'Encountered error while trying to delete git release endpoint'
+
+    def rollbackAutoCommits(self, n_autocommits=2):
+        info('Rolling back "%s" autocommits' % self.tag)
+
+        if self.tag not in self.local_repo.tags:
+            error('Could not find a SHA to rollback to for tag "%s"' % self.tag)
+            return False
+
+        # Search for auto release SHAs that match the release tag SHA
+        tag_commit = self.local_repo.tag('refs/tags/%s' % self.tag).commit
+        prior_commit = self.local_repo.commit(tag_commit.hexsha + '~%d' % n_autocommits)
+        n_commits_searched = 0
+        for commit in self.local_repo.iter_commits():
+            if n_commits_searched > 25:
```

Comment:
Why 25?

Reply:
It's arbitrary, but would fit better as an optional arg. Adding now

Reply:
It's arbitrary, refactoring to at least make it an optional arg

Reply:
done